silences-gitops/README.md

```diff
@@ -0,0 +1,114 @@
+
+Those Silences CRs are currently stored in a private Github repository at [https://github.com/giantswarm/silences][1], where GiantSwarm employees can create and manage silences. They are deployed to management clusters via a CronJob which pulls Silences CRs from the Github repository.
+
+## Problem statement
+
+The current deployment solution using a CronJob suffers from both privacy and security issues and does not allow customers to manage their silences. It is also not in line with any of the deployment mechanisms we currently have in place.
+
+## Solution proposal
+
+This solution proposes to re-use the current [management-cluster-base][2] (MCB) and customer management cluster (CMC) (e.g. [giantswarm-management-clusters](https://github.com/giantswarm/giantswarm-management-clusters)) GitOps mechanism in place in order to manage and deploy Silences CRs accross management clusters.
+
+This solution is laid out as follows:
+
+- Silence CRs common to all clusters are stored in the MCB repository, under the `bases/silences/` directory
+- Silence CRs added in MCB also need to be referenced in the `bases/silences/kustomization.yaml` file
+- Silence CRs specific to a cluster are stored in the corresponding CMC repository, under the `management-clusters/<management-cluster-name>/silences/` directory
+- Name collision between MCB and CMC Silences CRs is prevented by adding the special `global-` name prefix when deploying Silences CRs from the MCB repository
+
+![Silences GitOps Architecture](silences-gitops.png)
+
+### Examples
+
+- Common Silence CR in MCB repository:
+
+```yaml
+$ cat management-cluster-bases/bases/silences/alice.yaml
+apiVersion: monitoring.giantswarm.io/v1alpha1
+kind: Silence
+metadata:
+  name: alice
+  annotations:
+    valid-until: "2100-01-01"
+spec:
+  matchers:
+  - name: alertname
+    value: AliceAlert
+    isRegex: false
+```
+
+```yaml
+$ cat management-cluster-bases/bases/silences/kustomization.yaml
+resources:
+- alice.yaml
+```
+
+- Cluster specific Silence CR in CMC repository:
+
+```yaml
+$ cat giantswarm-management-clusters/management-clusters/some-management-cluster/silences/bob.yaml
+apiVersion: monitoring.giantswarm.io/v1alpha1
+kind: Silence
+metadata:
+  name: bob
+  annotations:
+    valid-until: "2100-01-01"
+spec:
+  matchers:
+  - name: alertname
+    value: BobAlert
+    isRegex: false
+  - name: cluster_id
+    value: some-cluster-id
+    isRegex: false
+```
+
+### Use cases
+
+- Create a silence for all clusters
+  - Add the Silence CR to MCB repository in `bases/silences/` directory
+  - Reference the new Silence CR in the `bases/silences/kustomization.yaml` file
+
+- Create a silence for a single cluster
+  - Add the Silence CR to the `management-clusters/some-management-cluster/silences/` directory
+  - Add a matcher for the `cluster_id` tag with the cluster ID as value in the Silence CR
```

Comment:
Just a thought, to please avoid adding any `cluster_id` or user specific identifiers to silences in MCB. E.g. it could be tempting when you want to create silence across many customers and it would be "painful" to add to n CMC repos.

Another idea: does it make sense to have provider specific silences? We already have provider bases in MCB. That is pulled into the `flux` Kustomziation in all MC, so that kind of contradicts the separate KS for silences. Tho they do not have to be referenced in the provider k8s kustomization file and just added to each MC silences kustomization.yaml as a remote base as well.

Reply:
Can you give an example on how the provider based Silences CRs would work ?

Reply:
You simply extend the provider base with a silences folder, add a `kustomization.yaml` in there and reference it in the CMC `kustomization.yaml` that is reconciled by the `silences` Flux Kustomization. So the CMC one would look something like:

```yaml
resources:
  - https://github.com/giantswarm/management-cluster-bases//bases/silences?ref=main
  - https://github.com/giantswarm/management-cluster-bases//bases/provider/capa/flux-v2/silences?ref=main
```

Reply:
Taking this into consideration for the implementation.